Sources/Diagnostics/DiagnosticsTracker.swift

```diff
@@ -13,23 +13,45 @@
 
 import Foundation
 
-@available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
 protocol DiagnosticsTrackerType {
 
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
     func track(_ event: DiagnosticsEvent) async
 
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    func trackCustomerInfoVerificationResultIfNeeded(_ customerInfo: CustomerInfo,
+                                                     timestamp: Date) async
+
 }
 
 @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
 final class DiagnosticsTracker: DiagnosticsTrackerType {
 
-    private let diagnosticsFileHandler: DiagnosticsFileHandler
+    private let diagnosticsFileHandler: DiagnosticsFileHandlerType
 
-    init(diagnosticsFileHandler: DiagnosticsFileHandler) {
+    init(diagnosticsFileHandler: DiagnosticsFileHandlerType) {
         self.diagnosticsFileHandler = diagnosticsFileHandler
     }
 
     func track(_ event: DiagnosticsEvent) async {
         await diagnosticsFileHandler.appendEvent(diagnosticsEvent: event)
     }
+
+    func trackCustomerInfoVerificationResultIfNeeded(
+        _ customerInfo: CustomerInfo,
+        timestamp: Date = Date()
+    ) async {
+        let verificationResult = customerInfo.entitlements.verification
+        if verificationResult == .notRequested {
+            return
+        }
```

Comment:
I agree using guard for early returns, but yes, I think that as you say I prefer the if because of the negation. If you don't mind I will leave it with the `if` 😄

Reply:
Not at all, please do!